client/a8c-for-agencies/sections/sites/needs-setup-sites/site-configurations-modal/use-site-name.tsx

```diff
@@ -0,0 +1,78 @@
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useState } from 'react';
+import wpcom from 'calypso/lib/wp';
+
+const SUBDOMAIN_LENGTH_MINIMUM = 4;
+const SUBDOMAIN_LENGTH_MAXIMUM = 50;
+
+const checkSiteAvailability = async ( siteName: string ) => {
+	// To do: check site availability on the server
+	if ( siteName ) {
+		return true;
+	}
+};
+
+const getRandomSiteName = async () => {
+	const { suggestions } = await wpcom.req.get( {
+		apiNamespace: 'wpcom/v2',
+		path: `/site-suggestions`,
+	} );
+	const { title } = suggestions[ 0 ];
+
+	const { body: urlSuggestions } = await wpcom.req.get( {
+		apiNamespace: 'rest/v1.1',
+		path: `/domains/suggestions?http_envelope=1&query=${ title }&quantity=10&include_wordpressdotcom=true&include_dotblogsubdomain=false&only_wordpressdotcom=true&vendor=dot&managed_subdomain_quantity=0`,
+	} );
+
+	for ( const { domain_name } of urlSuggestions ) {
+		const siteName = domain_name.split( '.' )[ 0 ];
+		const isAvailable = await checkSiteAvailability( siteName );
+		if ( isAvailable ) {
+			return siteName;
+		}
+	}
+
+	// If all 10 urlSuggestions are taken (very unlikly) return empty '' name as falback
+	return '';
+};
+
+export const useSiteName = () => {
+	const translate = useTranslate();
+	const [ siteName, setSiteName ] = useState( '' );
+	const [ isRandomSiteLoading, setIsRandomSiteLoading ] = useState( true );
+
+	useEffect( () => {
+		getRandomSiteName()
+			.then( ( randomSiteName ) => {
+				setSiteName( randomSiteName );
+				setIsRandomSiteLoading( false );
+			} )
+			.catch( () => {
+				setSiteName( '' );
+				setIsRandomSiteLoading( false );
+			} );
+	}, [] );
+
+	let validationMessage: string | React.ReactNode;
```

Comment:
The `validationMessage` could be memoized to avoid calling translations on every render.

Reply:
Thank you, just addressed it.